database/postgres_store.go

```diff
@@ -0,0 +1,53 @@
+package database
+
+import (
+	"context"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/jmoiron/sqlx"
+	_ "github.com/lib/pq"
+	"time"
+)
+
+const (
+	connTimeOut = 10 * time.Second
+)
+
+type postgresStore struct {
+	DB *sqlx.DB
+}
+
+func NewPostgresStore(dsn string) *postgresStore {
+	db := sqlx.MustConnect("postgres", dsn)
+	return &postgresStore{
+		DB: db,
+	}
+}
+
+func (d *postgresStore) Close() {
+	d.DB.Close()
+}
+
+func (d *postgresStore) SaveRequestEntry(in *RequestEntry) error {
+	in.InsertedAt = time.Now()
+	query := `INSERT INTO rpc_endpoint_requests 
+	(id, received_at, inserted_at, request_duration_ms, is_batch_request, num_request_in_batch, http_method, http_url, http_query_param, http_response_status, ip_hash, origin, host, error) VALUES (:id, :received_at, :inserted_at, :request_duration_ms, :is_batch_request, :num_request_in_batch, :http_method, :http_url, :http_query_param, :http_response_status, :ip_hash, :origin, :host, :error)`
+	ctx, cancel := context.WithTimeout(context.Background(), connTimeOut)
+	defer cancel()
+	if _, err := d.DB.NamedExecContext(ctx, query, in); err != nil {
+		log.Error("[postgresStore] SaveRequestEntry failed", "id", in.Id, "error", err)
+		return err
+	}
+	log.Info("[postgresStore] SaveRequestEntry succeeded", "RequestEntry", in) // TODO:Remove logging
```

Comment:
logging should be done outside of db

Reply:
updated. Instead of logging inside each method, this will be logged from SaveRequest method